src/dispatch/DataPlaneClient.ts

```diff
@@ -114,6 +114,11 @@ export class DataPlaneClient {
         const options = {
             method: METHOD,
             protocol: this.config.endpoint.protocol,
+            port:
+                !this.config.endpoint.port ||
```

Comment:
Is there coverage for this line?

Not sure what the point of `!this.config.endpoint.port` is.